### ISSUE TITLE
Empty Flow as RS Processor TCK issue

### ISSUE THREAD

**Issue by ktoso** [akka/akka#24581]
@jroper found that required_spec109_mustIssueOnSubscribeForNonNullSubscriber sometimes (super rarely, so likely a concurrency issue) may fail with: 

```

[info]   java.lang.AssertionError: Async error during test execution: onSubscribe should be called prior to onComplete always
[info]   at org.testng.Assert.fail(Assert.java:78)
[info]   at org.reactivestreams.tck.TestEnvironment.verifyNoAsyncErrorsNoDelay(TestEnvironment.java:314)
[info]   at org.reactivestreams.tck.PublisherVerification$11.run(PublisherVerification.java:507)
[info]   at org.reactivestreams.tck.PublisherVerification.activePublisherTest(PublisherVerification.java:1138)
[info]   at org.reactivestreams.tck.PublisherVerification.required_spec109_mustIssueOnSubscribeForNonNullSubscriber(PublisherVerification.java:477)
[info]   at org.reactivestreams.tck.IdentityProcessorVerification.required_spec109_mustIssueOnSubscribeForNonNullSubscriber(IdentityProcessorVerification.java:311)
[info]   at akka.stream.tck.EmptyProcessorTest.reproduce(EmptyProcessorTest.scala:19)
[info]   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[info]   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
[info]   at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:74)
[info]   at org.testng.internal.Invoker.invokeMethod(Invoker.java:673)
[info]   at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:846)
[info]   at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1170)
[info]   at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
[info]   at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
[info]   at org.testng.TestRunner.runWorkers(TestRunner.java:1147)
[info]   at org.testng.TestRunner.privateRun(TestRunner.java:749)
[info]   at org.testng.TestRunner.run(TestRunner.java:600)
[info]   at org.testng.SuiteRunner.runTest(SuiteRunner.java:317)
[info]   at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:312)
[info]   at org.testng.SuiteRunner.privateRun(SuiteRunner.java:274)
[info]   at org.testng.SuiteRunner.run(SuiteRunner.java:223)
[info]   at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
[info]   at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
[info]   at org.testng.TestNG.runSuitesSequentially(TestNG.java:1039)
[info]   at org.testng.TestNG.runSuitesLocally(TestNG.java:964)
[info]   at org.testng.TestNG.run(TestNG.java:900)
[info]   at org.scalatest.testng.TestNGSuiteLike$class.run(TestNGSuiteLike.scala:261)
[info]   at akka.stream.tck.AkkaIdentityProcessorVerification.run(AkkaIdentityProcessorVerification.scala:16)
[info]   at org.scalatest.testng.TestNGSuiteLike$class.runTestNG(TestNGSuiteLike.scala:248)
[info]   at akka.stream.tck.AkkaIdentityProcessorVerification.runTestNG(AkkaIdentityProcessorVerification.scala:16)
[info]   at org.scalatest.testng.TestNGSuiteLike$class.run(TestNGSuiteLike.scala:149)
[info]   at akka.stream.tck.AkkaIdentityProcessorVerification.run(AkkaIdentityProcessorVerification.scala:16)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:480)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:300)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:290)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   at java.lang.Thread.run(Thread.java:748)
[info]   Cause: org.reactivestreams.tck.TestEnvironment$Latch$ExpectedClosedLatchException: onSubscribe should be called prior to onComplete always
[info]   at org.reactivestreams.tck.TestEnvironment$Latch.assertClosed(TestEnvironment.java:835)
[info]   at org.reactivestreams.tck.PublisherVerification$11$1.onComplete(PublisherVerification.java:503)
[info]   at akka.stream.impl.ReactiveStreamsCompliance$.tryOnComplete(ReactiveStreamsCompliance.scala:106)
[info]   at akka.stream.impl.VirtualProcessor.onComplete(StreamLayout.scala:201)
[info]   at akka.stream.scaladsl.Flow$$anonfun$toProcessor$1$$anon$1.onComplete(Flow.scala:280)
[info]   at org.reactivestreams.example.unicast.AsyncIterablePublisher$SubscriptionImpl.doSubscribe(AsyncIterablePublisher.java:148)
[info]   at org.reactivestreams.example.unicast.AsyncIterablePublisher$SubscriptionImpl.run(AsyncIterablePublisher.java:221)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   at java.lang.Thread.run(Thread.java:748)
```

For an empty processor, such as: `Flow[Int].toProcessor.run()`

We've always covered the Publisher side of this and that never failed, there may be something special about the processor we have there. 

Submitting reproducer.

**Comment by ktoso** [comment]
(it could be a race in tck or akka, not entirely sure yet btw)

**Comment by johanandren** [comment]
Potentially related #22981

**Comment by johanandren** [comment]
Created a repeat job here: https://jenkins.akka.io:8498/job/akka-local-repeat/8643/console

**Comment by 2m** [comment]
Recently this one failed in the Scala community build: https://github.com/scala/community-builds/pull/690#issuecomment-372831890

**Comment by johanandren** [comment]
Should be fixed by #24722

**Comment by ktoso** [comment]
Resolved, thanks a ton Johan
